### PR TITLE
2 new question form

### LINKED ISSUE #2: Form page for creating new questions
Use form system to create the question creation page

### PR DESCRIPTION
**Pull request by ljweston**
Closes #2 

Users can now ask a new question, add tags, delete tags, and publish a new question

### CODE REVIEW

File: src/Controller/QuestionController.php [hunk 1]
```diff
@@ -53,24 +60,55 @@ public function homepage(QuestionRepository $repository, int $page = 1)
     }
 
     /**
```
Comment (mlellis) [inline src/Controller/QuestionController.php]: Change this annotation out to the new style and use the Security method for the is_granted check

File: src/Controller/QuestionController.php [hunk 2]
```diff
@@ -53,24 +60,55 @@ public function homepage(QuestionRepository $repository, int $page = 1)
...
+            ]);
+        }
+        // no submittion, initial render 
+        return $this->render('questions/new.html.twig', [
+            'questionForm' => $form->createView(),
+        ]);
+    }
+
+    #[Route('/questions/publish/{slug}', name: "app_question_publish")]
+    public function publish(Question $question)
```
Comment (mlellis) [inline src/Controller/QuestionController.php:93]: Add a security check here

File: src/Controller/QuestionController.php [hunk 3]
```diff
@@ -53,24 +60,55 @@ public function homepage(QuestionRepository $repository, int $page = 1)
...
+    {
+        $dt = new DateTime();
+        $question->setAskedAt($dt);
+
+        $this->em->persist($question);
```
Comment (mlellis) [inline src/Controller/QuestionController.php]: You don't need to persist this. You only need that when you are creating a new instance - not when it exists already.

File: templates/questions/show.html.twig [hunk 4]
```diff
@@ -34,6 +34,24 @@
                                     </div>
                                 </form>
                             </div>
+                            {# Remove tags #}
+                            {% if is_granted('EDIT', question) %}
+                                <form action="{{ path('app_question_publish', { slug: question.slug }) }}" method="POST">
+                                    <div class="mt-2">
+                                        <h3>Tags:</h3>
```
Comment (mlellis) [inline templates/questions/show.html.twig]: Move all of these tag links outside of the form.

File: templates/questions/show.html.twig [hunk 5]
```diff
@@ -34,6 +34,24 @@
...
+                                        {% for questionTag in question.questionTags %}
+                                            <a href="{{ path('app_question_remove_tag', {
+                                                id: questionTag.id,
+                                            }) }}" class="badge rounded-pill bg-light text-dark">
+                                            {{ questionTag.tag }}
+                                            </a>
+                                        {% endfor %}
+                                    </div>
+                                    {% if question.askedAt == null %}
```
Comment (mlellis) [inline templates/questions/show.html.twig]: I think the form should live within this conditional check

**Review by mlellis — CHANGES_REQUESTED** [review]

File: src/Controller/QuestionController.php (hunk 1 above)
Comment (ljweston) [reply src/Controller/QuestionController.php]: Done

File: src/Controller/QuestionController.php (hunk 2 above)
Comment (ljweston) [reply src/Controller/QuestionController.php:93]: Done

File: src/Controller/QuestionController.php (hunk 3 above)
Comment (ljweston) [reply src/Controller/QuestionController.php]: Done

File: templates/questions/show.html.twig (hunk 4 above)
Comment (ljweston) [reply templates/questions/show.html.twig]: Done

File: templates/questions/show.html.twig (hunk 5 above)
Comment (ljweston) [reply templates/questions/show.html.twig]: Done

File: templates/questions/show.html.twig [hunk 6]
```diff
@@ -46,11 +45,12 @@
...
                                             </a>
                                         {% endfor %}
-                                    </div>
-                                    {% if question.askedAt == null %}
-                                        <button class="mt-5 btn btn-primary">Publish</button>
-                                    {% endif %}
-                                </form>                                
+                                </div>
+                                {% if question.askedAt == null %}
+                                <form action="{{ path('app_question_publish', { slug: question.slug }) }}" method="POST">
```
Comment (mlellis) [inline templates/questions/show.html.twig]: After looking at this again, I don't think you need a form at all here. You should be able to just have a link to this action. The post request isn't actually doing anything.

Also, I can't tell for sure, but it looks like the formatting might be off here. Just double check the indentations.

**Review by mlellis — CHANGES_REQUESTED** [review]

File: templates/questions/show.html.twig (hunk 6 above)
Comment (ljweston) [reply templates/questions/show.html.twig]: Checked formatting and removed form, replaced button with anchor tags

**Review by mlellis — APPROVED** [review]

**Pull request merged**